cli/cmd/uninstall/uninstall.go

```diff
@@ -0,0 +1,439 @@
+	if err != nil {
+		return false, "", "", fmt.Errorf("error finding existing installations: %s", err)
+	}
+
+	found := false
+	foundReleaseName := ""
+	foundReleaseNamespace := ""
+	for _, rel := range res {
+		if rel.Chart.Metadata.Name == "consul" {
+			if c.flagNamespace != DefaultAllNamespaces {
+				if c.flagNamespace == rel.Name {
+					found = true
+					foundReleaseName = rel.Name
+					foundReleaseNamespace = rel.Namespace
+					break
+				}
+			}
+			found = true
+			foundReleaseName = rel.Name
+			foundReleaseNamespace = rel.Namespace
+			break
+		}
+	}
+
+	return found, foundReleaseName, foundReleaseNamespace, nil
+}
+
+func (c *Command) deletePVCs(foundReleaseName, foundReleaseNamespace string) error {
+	var pvcNames []string
+	pvcSelector := metav1.ListOptions{LabelSelector: fmt.Sprintf("release=%s", foundReleaseName)}
+	pvcs, err := c.kubernetes.CoreV1().PersistentVolumeClaims(foundReleaseNamespace).List(c.Ctx, pvcSelector)
+	if err != nil {
+		return fmt.Errorf("deletePVCs: %s", err)
+	}
+	for _, pvc := range pvcs.Items {
+		err := c.kubernetes.CoreV1().PersistentVolumeClaims(foundReleaseNamespace).Delete(c.Ctx, pvc.Name, metav1.DeleteOptions{})
+		if err != nil {
+			return fmt.Errorf("deletePVCs: error deleting PVC %q: %s", pvc.Name, err)
+		}
+		pvcNames = append(pvcNames, pvc.Name)
+	}
+	err = backoff.Retry(func() error {
+		pvcs, err := c.kubernetes.CoreV1().PersistentVolumeClaims(foundReleaseNamespace).List(c.Ctx, pvcSelector)
+		if err != nil {
+			return fmt.Errorf("deletePVCs: %s", err)
+		}
+		if len(pvcs.Items) > 0 {
+			err = fmt.Errorf("deletePVCs: pvcs still exist")
+			return err
+		}
+		return err
+	}, backoff.WithMaxRetries(backoff.NewConstantBackOff(100*time.Millisecond), 1800))
+	if err != nil {
+		return fmt.Errorf("deletePVCs: timed out waiting for PVCs to be deleted")
+	}
+	if len(pvcNames) > 0 {
+		c.UI.Output(common.PrefixLines(" Deleted PVC => ", strings.Join(pvcNames, "\n")), terminal.WithSuccessStyle())
+		c.UI.Output("PVCs deleted.", terminal.WithSuccessStyle())
+	} else {
+		c.UI.Output("No PVCs found.", terminal.WithSuccessStyle())
+	}
+	return nil
+}
+
+// deleteSecrets deletes any secrets that have foundReleaseName in their name.
+func (c *Command) deleteSecrets(foundReleaseName, foundReleaseNamespace string) error {
+	var secretNames []string
+	secrets, err := c.kubernetes.CoreV1().Secrets(foundReleaseNamespace).List(c.Ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("deleteSecrets: %s", err)
+	}
+	for _, secret := range secrets.Items {
+		if strings.HasPrefix(secret.Name, foundReleaseName) {
+			err := c.kubernetes.CoreV1().Secrets(foundReleaseNamespace).Delete(c.Ctx, secret.Name, metav1.DeleteOptions{})
+			if err != nil {
+				return fmt.Errorf("deleteSecrets: error deleting Secret %q: %s", secret.Name, err)
+			}
+			secretNames = append(secretNames, secret.Name)
+		}
+	}
+	if len(secretNames) > 0 {
+		c.UI.Output("Consul secrets deleted.", terminal.WithSuccessStyle())
```

Comment:
This is a super nit comment, but if there are 3 secrets and 2 got deleted and the third exits the for loop because you cant delete it for some reason then neither of these Outputs get processed properly. I suppose it is okay since it "should work" in happy path.

Reply:
Maybe the correct thing to do is output the Error via `c.UI.Output()` instead of return the error, and then continue to process the other secrets?

I guess this logic applies to the other similar functions, if you choose.

Reply:
I think its ok because in an error case, it will exit the function at the point of the error without printing either of these outputs here, and just print the first error it got to. Then when a user re-runs uninstall after fixing their issue, it should succeed and delete the secrets properly.